lib/model/model.go

```diff
@@ -1242,65 +1241,55 @@ func (m *Model) ConnectedTo(deviceID protocol.DeviceID) bool {
 }
 
 func (m *Model) GetIgnores(folder string) ([]string, []string, error) {
-	var lines []string
-
 	m.fmut.RLock()
 	cfg, ok := m.folderCfgs[folder]
 	m.fmut.RUnlock()
 	if !ok {
-		return lines, nil, fmt.Errorf("Folder %s does not exist", folder)
-	}
-
-	if !cfg.HasMarker() {
-		return lines, nil, fmt.Errorf("Folder %s stopped", folder)
-	}
-
-	fd, err := os.Open(filepath.Join(cfg.Path(), ".stignore"))
-	if err != nil {
-		if os.IsNotExist(err) {
-			return lines, nil, nil
+		cfg, ok = m.cfg.Folders()[folder]
+		if ok {
+			matcher := ignore.New(false)
+			path := filepath.Join(cfg.Path(), ".stignore")
+			if err := matcher.Load(path); err != nil {
+				return nil, nil, err
+			}
+			return matcher.Lines(), matcher.Patterns(), nil
 		}
-		l.Warnln("Loading .stignore:", err)
-		return lines, nil, err
+		return nil, nil, fmt.Errorf("Folder %s does not exist", folder)
 	}
-	defer fd.Close()
 
-	scanner := bufio.NewScanner(fd)
-	for scanner.Scan() {
-		lines = append(lines, strings.TrimSpace(scanner.Text()))
+	if !cfg.HasMarker() {
+		return nil, nil, fmt.Errorf("Folder %s stopped", folder)
 	}
 
 	m.fmut.RLock()
+	lines := m.folderIgnores[folder].Lines()
 	patterns := m.folderIgnores[folder].Patterns()
 	m.fmut.RUnlock()
 
 	return lines, patterns, nil
 }
 
 func (m *Model) SetIgnores(folder string, content []string) error {
+	m.fmut.RLock()
 	cfg, ok := m.folderCfgs[folder]
+	m.fmut.RUnlock()
+	var paused bool
 	if !ok {
-		return fmt.Errorf("Folder %s does not exist", folder)
+		cfg, paused = m.cfg.Folders()[folder]
```

Comment:
It is used at the end of the function and it means paused, i.e. folder exists but is not currently in the model. In case a folder is unpaused, the new ignores are written to ".stignore" and then a rescan is scheduled to pick up the new ignores and the resulting changes to the index. This cannot happen when a folder is paused.